mne/viz/backends/_notebook.py

```diff
@@ -171,6 +171,26 @@ def _tool_bar_add_text(self, name, value, placeholder):
     def _tool_bar_add_spacer(self):
         pass
 
+    def _tool_bar_add_screenshot_button(self, name, desc, func):
+        def _screenshot():
+            from PIL import Image
+            fname = self.actions[f"{name}_field"].value
+            fname = self._get_screenshot_filename() \
+                if len(fname) == 0 else fname
+            img = func()
+            Image.fromarray(img).save(fname)
```

Comment:
This code is not DRY. We already have a `_ndarray_to_fig(img).savefig(filename)` in `_pysurfer_mayavi`, maybe we should instead have a `_save_ndarray_img(fname, img)` somewhere that is used by `_notebook.py`, `_qt.py`, and `_pysurfer_mayavi.py`?

Reply:
I did it in d6a6fc7